airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java

```diff
@@ -374,7 +374,7 @@ public String softReset(final StreamConfig stream) {
     final String clearLoadedAt = clearLoadedAt(stream.id());
     // We just unset loaded_at on all raw records, so we need to process all of them (i.e. should not
     // filter on extracted_at)
-    final String rebuildInTempTable = updateTable(stream, SOFT_RESET_SUFFIX, Optional.empty());
+    final String rebuildInTempTable = updateTable(stream, SOFT_RESET_SUFFIX, Optional.empty(), true);
```

Comment:
Why do soft-resets always use the more expensive version? Shouldn't this also be the try/catch?

Reply:
we might need to refactor the softReset method to live in the TyperDeduper, and promote the `clearLoadedAt` method to a public method on SqlGenerator. I.e. in DefaultTyperDeduper:

```java
private void softReset(StreamConfig stream) {
  // This method doesn't currently exist in the SqlGenerator interface, it's a private method in BigQuerySqlGenerator
  execute(generator.clearLoadedAt(stream));
  execute(generator.createTable(stream, "_soft_reset"));
  try {
    execute(generator.updateTable(stream, "_soft_reset", false));
  } catch (Exception e) {
    execute(generator.updateTable(stream, "_soft_reset", true));
  }
  execute(generator.overwriteFinalTable(stream, "_soft_reset"));
}
````

this try-catch can't live inside the sqlgenerator b/c sqlgenerator is side-effect free. i.e. it can't actually execute sql

Reply:
+1 to ed's comment - I think its ok to leave it out for now, soft resets are erring on the side of being the expensive but reliable query but we should probably extract it

Reply:
I actually just realized - we probably should do this refactor before releasing this pr? Otherwise this will be a regression in bigquery soft reset performance

Reply:
open to discussion though. maybe that's worth the improvement for query length

Reply:
fair enough! refactored